### DISCUSSION TITLE
go: go.mod requires go >= 1.23.1 (running go 1.22.7; GOTOOLCHAIN=local)

### DISCUSSION THREAD

**Discussion by JCzz** [dagger/dagger#12207] [Help]
### What is the issue?

Hi

From this: https://docs.dagger.io/manuals/user/chaining

Running this func:
`dagger -m github.com/kpenfound/dagger-modules/golang@v0.2.0 call build --source=https://github.com/dagger/dagger --args=./cmd/dagger entries
`
```
Error: response from query: input: golang.build resolve: call function "Build": process "/runtime" did not complete successfully: exit code: 2

Stdout:
marshal: json: error calling MarshalJSON for type *dagger.Directory: input: container.from.withMountedCache.withMountedCache.withDirectory.withWorkdir.withEnvVariable.withEnvVariable.withExec.directory resolve: process "go build -o /out/ ./cmd/dagger" did not complete successfully: exit code: 1

Stderr:
go: go.mod requires go >= 1.23.1 (running go 1.22.7; GOTOOLCHAIN=local)
```

### Dagger version

dagger version dagger v0.13.0 (registry.dagger.io/engine:v0.13.0) darwin/arm64

**Comment by jedevc** [comment]
Ah, this is an issue that @kpenfound's module uses an old version of go (while `github.com/dagger/dagger` pretty much tracks go releases).

@dagger/docs could we update this module to use a more modern version of go? 

The chaining example doesn't currently work.
